### PR TITLE
fix test instability

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Fix test instability
# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: tests/deploy/e2e/test_autostarted.py [hunk 1]
```diff
@@ -233,31 +233,23 @@ async def test_deploy_no_code(resource_container, client, clienthelper, environm
         }
     ]
 
-    await clienthelper.put_version_simple(resources, version)
-
-    await clienthelper.wait_for_deployed(version)
-    # The resource state and its logs are not set atomically. This call prevents a race condition.
-    await wait_until_logs_are_available(client, environment, resource_id, expect_nr_of_logs=3)
-
```
Comment (wouterdb) [inline tests/deploy/e2e/test_autostarted.py:241]: This waiting was brittle, I now wait for the final condition alone

File: tests/deploy/e2e/test_autostarted.py [hunk 2]
```diff
@@ -233,31 +233,23 @@ async def test_deploy_no_code(resource_container, client, clienthelper, environm
...
-    result = response.result
-    assert result["resource"]["status"] == "unavailable"
+    await clienthelper.put_version_simple(resources, version, wait_for_released=True)
 
-    logging.getLogger(__name__).warning("Found results: %s", json.dumps(result["logs"], indent=1))
+    async def log_any() -> bool:
+        response = await client.get_resource(environment, resource_id, logs=True)
+        assert response.code == 200
+        result = response.result
+        logging.getLogger(__name__).warning("Found results: %s", json.dumps(result["logs"], indent=1))
```
Comment (Hugo-Inmanta) [inline tests/deploy/e2e/test_autostarted.py:242]:
```suggestion
```

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

**Review by Hugo-Inmanta — APPROVED** [review]
The warning debug log line should probably be removed, approving already still.

File: tests/deploy/e2e/test_autostarted.py (hunk 2 above)
Comment (wouterdb) [reply tests/deploy/e2e/test_autostarted.py:242]: it is very handy when the testcase fails,....

**Pull request closed without merging**